board/aspeed/evb_ast2700/evb_ast2700.c

```diff
@@ -186,6 +186,32 @@ int get_cia_ser_num(const uint8_t* fru_buf, char* chassis_ser_num)
 	return csn_len;
 }
 
+int get_csn_last4(const char* str, char* buf, size_t buf_len)
+{
+	int pos_min = 0;
+ 	int pos_max = 0;
+ 	int cur_pos = 0;
+ 	int c=0;
+ 	pos_max=strlen(str);
+ 
+ 	/* Trim leading, trailing whitespaces */
+ 	while(isspace(*(str+pos_min))) pos_min++;
+ 	while(isspace(*(str+pos_max-1))) pos_max--;
+ 	/* extract from end */
```

Comment:
// Ensure buffer is large enough
    if (buf_len < MIN_CSN_UNIQ_STR_LEN + 1) return -1;